http.go

```diff
@@ -0,0 +1,645 @@
+package algeneva
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/textproto"
+	"net/url"
+	"strings"
+)
+
+/*
+   RFC spec can be found at https://datatracker.ietf.org/doc/html/rfc7230
+   Syntax notation used in this file:
+       OWS    = optional whitespace
+       SP     = space
+       HTAB   = horizontal tab
+       CR     = carriage return
+       LF     = line feed
+       CRLF   = CR LF
+
+       ALPHA  = A-Z / a-z
+       DIGIT  = 0-9
+       VCHAR  = any visible ASCII character
+
+   Note: There are additional syntax notations defined in the RFC spec, but they are not used in
+	this file.
+*/
+
+// WriteRequest writes req to w while applying the strategy.
+func WriteRequest(w io.Writer, req *http.Request, strategy *HTTPStrategy) error {
+	w = newWriter(w, strategy)
+	return req.Write(w)
+}
+
+// writer is an io.Writer wrapper that applies a strategy to the request before writing it to the
+// wrapped writer. writer forces a flush after every write.
+type writer struct {
+	w               *bufio.Writer
+	buf             *bytes.Buffer
+	strategy        *HTTPStrategy
+	appliedStrategy bool
+}
+
+func newWriter(w io.Writer, strategy *HTTPStrategy) *writer {
+	bw, ok := w.(*bufio.Writer)
+	if !ok {
+		bw = bufio.NewWriter(w)
+	}
+
+	return &writer{
+		w:        bw,
+		buf:      new(bytes.Buffer),
+		strategy: strategy,
+	}
+}
+
+func (w *writer) Write(p []byte) (n int, err error) {
+	defer func() {
+		if n > 0 {
+			ferr := w.w.Flush()
+			if ferr != nil && err == nil {
+				err = ferr
+			}
+		}
+	}()
+
+	if w.appliedStrategy {
+		return w.w.Write(p)
+	}
+	i := bytes.Index(p, []byte("\r\n\r\n"))
+	if i == -1 {
+		w.buf.Write(p)
+		return len(p), nil
```

Comment:
In this case, can we return the buf Write response directly? Or we shouldn't return an error if it happens?
```suggestion
		return w.buf.Write(p)
```